src/arrow-util/src/builder.rs

```diff
@@ -156,6 +158,12 @@ fn scalar_to_arrow_datatype(scalar_type: &ScalarType) -> Result<DataType, anyhow
             }
         }
         ScalarType::String => DataType::LargeUtf8,
+        // Parquet does have a UUID 'Logical Type' in parquet format 2.4+, but there is no arrow
+        // UUID type, so for now we represent UUIDs as utf8-encoded strings.
+        ScalarType::Uuid => DataType::Utf8,
```

Comment:
Okay, so maybe the smarter thing to do would be to export these as 16-byte fixed-size arrays, just without the UUID Logical Type annotation until that gets supported in arrow. Then if the user can't parse these they can opt to cast the UUID column to text if they want to. I'll make that change - thanks for pointing this out!

Reply:
That sounds like a good idea to me! Apologies if you've already mentioned this somewhere and I missed it, but we could (should?) also annotate each field with an [extension type](https://github.com/MaterializeInc/materialize/issues/26796) e.g. `('ARROW:extension:name', 'materialize.v1.uuid')`. These should get included in Parquet and it would make it easier for folks to read back and correctly parse the resulting files.

Reply:
Pushed a commit with the additional metadata! Also switched mztimestamp types to use a UInt64 array instead as you suggested